### ISSUE TITLE
Paramspec Invariance Error

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#2531]
**Describe the bug**
I have a class that holds a callable and want to preserve the type information of its arguments. So I thought to use paramspec and have a function to convert the callable into that class. I'm running into error about type variance for paramspec and unsure what intended way to write this is. The error's also unclear to me as I get,

`Type "P@transform_foo" cannot be assigned to type "P@transform_foo"`

**To Reproduce**

```python
from typing import Callable, Generic, TypeVar

from typing_extensions import ParamSpec

P = ParamSpec("P")
R = TypeVar("R")


class Foo(Generic[P, R]):
    def __init__(self, func: Callable[P, R]):
        self.func = func


def transform_foo(f: Callable[P, R]) -> Foo[P, R]:
    return Foo(f)
```

Error message appears on last line of,

```
Expression of type "Foo[P@transform_foo, R@transform_foo]" cannot be assigned to return type "Foo[P@transform_foo, R@transform_foo]"
  TypeVar "P@Foo" is invariant
    Type "P@transform_foo" cannot be assigned to type "P@transform_foo" reportGeneralTypeIssues
```

**Expected behavior**
No errors.

**VS Code extension or command-line**
Both. pyright 1.1.184 and pylance v2021.11.0

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by Tomaz-Vieira** [comment]
I just had the same problem. Oddly enough, if you are explicit with the type arguments when creating `Foo`, that is, `return Foo[P, R](f)`, pyright will be happy. So maybe it's some inference problem?

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.185, which I just published. It will also be included in the next release of pylance.
